profile/src/identified_vec_via.rs

```diff
@@ -0,0 +1,262 @@
+use identified_vec::{
+    identified_vec_into_iterator::IdentifiedVecIntoIterator, Identifiable,
+    IdentifiedVecOf, IsIdentifiableVecOfVia, IsIdentifiedVec,
+    IsIdentifiedVecOf, ViaMarker,
+};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    fmt::{Debug, Display, Formatter},
+    ops::Index,
+};
+use uniffi::{
+    check_remaining,
+    deps::bytes::{Buf, BufMut},
+    metadata, Lift, Lower, LowerReturn, MetadataBuffer, RustBuffer,
+};
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct IdentifiedVecVia<Element: Identifiable + Debug + Clone> {
+    id_vec: IdentifiedVecOf<Element>,
```

Comment:
Ah I found the [`delegate` crate](https://crates.io/crates/delegate) after I wrote this, which seems excellent and can simplify/replace/clean this up a bit, and/or also use `Deref` / `DerefMut`!